[ZoneTelechargementBridge] Fix SSL errors

sysadminstory opened the pull request:
The streaming unprotected URI had some SSL issue because the domain is
now hosting a parking website. This URL is the only known URL that allows to
show streaming links without being protected by Cloudflare DDoS protection.

This URL still works if you ask Cloudflare for the content. Therefore,
to load this URL, this bridge kindly ask cURL to connect to the same IP
as the unproteced DDL URI to get the Streaming URI content, by using the
CURLOPT_RESOLVE option.

So we are still able to get streamingg links ! (fix for #2419)

sysadminstory committed:
[ZoneTelechargementBridge] Fix SSL errors

The streaming unprotected URI had some SSL issue because the domain is
now hosting a parking website. This URL is the only known URL that allows to
show streaming links without being protected by Cloudflare DDoS protection.

This URL still works if you ask Cloudflare for the content. Therefore,
to load this URL, this bridge kindly ask cURL to connect to the same IP
as the unproteced DDL URI to get the Streaming URI content, by using the
CURLOPT_RESOLVE option.

So we are still able to get streamingg links ! (fix for #2419)

sysadminstory committed:
[ZoneTelechargementBridge] Fix content links

The unprotected URLs uses the old link protection tool that is broken.
The links are rewritten to use the new link protection tool !

em92 commented:
cUrl error: error:14094410:SSL routines:ssl3_read_bytes:sslv3 alert handshake failure (35)
Do you reproduce it on your side? https://feed.eugenemolotov.ru/pr2426/?action=display&bridge=ZoneTelechargement&context=Suivre+la+publication+des+%C3%A9pisodes+d%27une+s%C3%A9rie+en+cours+de+diffusion&url=telecharger-series%2F31079-halt-and-catch-fire-saison-4-french-hd720p.html&filter=both&format=Html

sysadminstory committed:
[ZoneTelechargementBridge] Switch to another way to bypass SSL issues

Switch back to some existing unprotected URL instead of an URL using an
expired domain !

sysadminstory commented:
> cUrl error: error:14094410:SSL routines:ssl3_read_bytes:sslv3 alert handshake failure (35) Do you reproduce it on your side? https://feed.eugenemolotov.ru/pr2426/?action=display&bridge=ZoneTelechargement&context=Suivre+la+publication+des+%C3%A9pisodes+d%27une+s%C3%A9rie+en+cours+de+diffusion&url=telecharger-series%2F31079-halt-and-catch-fire-saison-4-french-hd720p.html&filter=both&format=Html

Yes, I reproduced it on my side too. I should not use this solution to bypass the SSL issue : using an expired domain to access a website using an HTTPS connection would have fail soon or later.

I switched back to simple solution using a domain which is not protected (as for now !)

em92 commented:
gj!
